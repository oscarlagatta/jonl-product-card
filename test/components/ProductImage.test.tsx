import { create } from 'react-test-renderer';
import React from 'react';
import { ProductCard, ProductImage } from '../../src';

import { product2 } from '../data/products';

describe('Product Image', () => {
  test('must show the component with custom image', () => {

    const wrapper = create(
      <ProductImage img="https://hello.jpg" className="custom-class" />
    );
    expect(wrapper.toJSON()).toMatchSnapshot();

  });

  test('Must show the component with the product image', () => {
    const wrapper = create(
      <ProductCard product={product2}>
        {
          () => (
            <ProductImage />
          )
        }
      </ProductCard>
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
})