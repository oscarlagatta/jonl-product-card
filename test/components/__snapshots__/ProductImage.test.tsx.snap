// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Image Must show the component with the product image 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product"
    className="productImg undefined"
    src="./coffee-mug2.png"
  />
</div>
`;

exports[`Product Image must show the component with custom image 1`] = `
<img
  alt="Product"
  className="productImg custom-class"
  src="https://hello.jpg"
/>
`;
